public/pages/Administration/pages/ManageWebhooks.page.tsx

```diff
@@ -0,0 +1,180 @@
+import React from "react"
+import { Button } from "@fider/components"
+
+import { Webhook, WebhookStatus, WebhookType } from "@fider/models"
+import { actions, Failure } from "@fider/services"
+import { AdminBasePage } from "../components/AdminBasePage"
+import { WebhookForm, WebhookFormState } from "../components/webhook/WebhookForm"
+import { WebhookListItem } from "../components/webhook/WebhookListItem"
+import { VStack } from "@fider/components/layout"
+
+interface ManageWebhooksPageProps {
+  webhooks: Webhook[]
+}
+
+interface ManageWebhooksPageState {
+  isAdding: boolean
+  allWebhooks: Webhook[]
+  editing?: Webhook
+}
+
+const webhookSorter = (w1: Webhook, w2: Webhook) => {
+  if (w1.name < w2.name) {
+    return -1
+  } else if (w1.name > w2.name) {
+    return 1
+  }
+  return 0
+}
+
+interface WebhooksListProps {
+  title: string
+  description: string
+  list: JSX.Element[]
+}
+
+const WebhooksList = (props: WebhooksListProps) => {
+  return (
+    <div>
+      <h2 className="text-display">{props.title}</h2>
+      <p className="text-muted">These webhooks are triggered every time {props.description}.</p>
+      <VStack spacing={4} divide={true}>
+        {props.list.length === 0 ? <p className="text-muted">There aren’t any &quot;{props.title.toLowerCase()}&quot; webhook yet.</p> : props.list}
+      </VStack>
+    </div>
+  )
+}
+
+export default class ManageWebhooksPage extends AdminBasePage<ManageWebhooksPageProps, ManageWebhooksPageState> {
```

Comment:
same here, if you could cover to function components please (see GeneralSettingsPage as an example to replace class inheritance)

Reply:
I'll check this to help me make this inheritance, I have no idea about what to do it right now :o
By the way, shouldn't we deprecate the elements we're trying to move away? It may avoid theses troubles next time